dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -354,6 +354,31 @@ InsertItemToListView(
     return -1;
 }
 
+BOOL tmToStr(
+    IN struct tm *pTM,
+    OUT WCHAR *szBuffer)
+{
+    SYSTEMTIME st;
+    WCHAR szBufferDate[100];
+    WCHAR szBufferTime[100];
+
+    st.wYear = pTM->tm_year + 1900;
+    st.wMonth = pTM->tm_mon + 1;
+    st.wDay = pTM->tm_mday;
+    st.wHour = pTM->tm_hour;
+    st.wMinute = pTM->tm_min;
+    st.wSecond = pTM->tm_sec;
+
+    if (GetDateFormatW(LOCALE_USER_DEFAULT, 0, &st, nullptr, szBufferDate, _countof(szBufferDate)) &&
+        GetTimeFormatW(LOCALE_USER_DEFAULT, 0, &st, nullptr, szBufferTime, _countof(szBufferTime)))
+    {
+        wcsncpy(szBuffer, szBufferDate, 100);
```

Comment:
Maybe limit szBuffer to 75 to make room for the concatenated buffers after and avoid buffer overflow ?

Reply:
ok